src/components/PhotoSwipe/PhotoSwipe.tsx

```diff
@@ -527,6 +543,100 @@ function PhotoSwipe(props: Iprops) {
         }
     }, [showInfo]);
 
+    useEffect(() => {
+        if (!isOpen) return;
+        const item = items[photoSwipe?.getCurrentIndex()];
+        if (item && item.metadata.fileType === FILE_TYPE.LIVE_PHOTO) {
+            let videoPlaying = false;
+
+            const playVideo = async (livePhotoVideo, livePhotoImage) => {
+                if (videoPlaying) return;
+                videoPlaying = true;
+                livePhotoVideo.style.opacity = 1;
+                livePhotoImage.style.opacity = 0;
+                livePhotoVideo.load();
+                livePhotoVideo.play().catch(() => {
+                    pauseVideo(livePhotoVideo, livePhotoImage);
+                });
+            };
+
+            const pauseVideo = async (livePhotoVideo, livePhotoImage) => {
+                if (!videoPlaying) return;
+                videoPlaying = false;
+                livePhotoVideo.pause();
+                livePhotoVideo.style.opacity = 0;
+                livePhotoImage.style.opacity = 1;
+            };
+            setLivePhotoBtn(
+                <LivePhotoBtn
```

Comment:
I think instead of updating the complete button we can just update the eventHandler functions?

something like `fav button`. have a state isLivePhoto state which would update whenever the currentItem change and that could handle the hiding and showing of the button and just pass the current item id to play and pause Video handlers so that they could construct the className and select the element and trigger the required action

Reply:
Sure, will do this.